From fe074a010d3eaaee4fff72693a779cc4d4361ce5 Mon Sep 17 00:00:00 2001
From: Vasileios Bimpikas <vasileios.bimpikas@xilinx.com>
Date: Mon, 3 Dec 2018 13:58:50 +0000
Subject: [PATCH 1151/1852] staging: xroeframer: Add devicetree binding

commit e3f312fae581d521be047ccf9adf9d7fe25987b0 from
https://github.com/Xilinx/linux-xlnx.git

Adding devicetree binding entry under
Documentation/devicetree/bindings/staging/xroeframer. Removing
relevant TODO from main source file (xroe_framer.c)

Signed-off-by: Vasileios Bimpikas <vasileios.bimpikas@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/staging/xroeframer/dt-binding.txt | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 drivers/staging/xroeframer/dt-binding.txt

diff --git a/drivers/staging/xroeframer/dt-binding.txt b/drivers/staging/xroeframer/dt-binding.txt
new file mode 100644
index 000000000000..8dabef16d083
--- /dev/null
+++ b/drivers/staging/xroeframer/dt-binding.txt
@@ -0,0 +1,17 @@
+* Xilinx Radio over Ethernet Framer driver
+
+Required properties:
+- compatible: must be "xlnx,roe-framer-1.0"
+- reg: physical base address of the framer and length of memory mapped region
+- clock-names: list of clock names
+- clocks: list of clock sources corresponding to the clock names
+
+Example:
+	roe_framer@a0000000 {
+		compatible = "xlnx,roe-framer-1.0";
+		reg = <0x0 0xa0000000 0x0 0x10000>;
+		clock-names = "s_axi_aclk", "m_axis_defm_aclk",
+			      "s_axis_fram_aclk", "tx0_eth_port_clk",
+			      "internal_bus_clk";
+		clocks = <0x43 0x44 0x44 0x45 0x45>;
+	};
-- 
2.31.1

